### ISSUE TITLE
Galaxy image metadata framework needs enhancements

### ISSUE THREAD

**Issue by gregvonkuster** [galaxyproject/galaxy#659]
In addition to image type, image metadata should include (but not limited to):
- Bit depth (8, 16, 24, 32)
- Whether the image is binary
- Light or dark background with respect to foreground
- Settings made within previous tools that could impact selections made in following tool runs

I'm wondering if enhancements to image metadata are planned for the near future.  If not, I would like to implement some enhancements, but I'm wondering if I can count on PIL being available for use or if I should plan to import something else.  Currently, PIL cannot be expected to exist for image metadata, but if I can require it I think it will provide all that I need for these enhancements.

**Comment by dannon** [comment]
I don't know of any image metadata enhancements currently planned.  Just regarding libraries -- the original PIL, no, but https://python-pillow.github.io/ (which is a drop-in replacement) we could probably package and ship if need be.

**Comment by gregvonkuster** [comment]
Yes, the key for me is whether I can expect to do something like this:

`from PIL import Image`

which would replace the current:

```
try:
    import Image as PIL
except ImportError:
    try:
        from PIL import Image as PIL
    except:
        PIL = None
```

**Comment by dannon** [comment]
Yep, I'd think we could do that if it was going to be useful to folks.

**Comment by gregvonkuster** [comment]
Awesome!  So, what would the process be to make https://python-pillow.github.io/ available to the Galaxy framework?

**Comment by dannon** [comment]
Depending on how fast you need/want to get this in, I'd say we should get #428 completed and merged, and then it should be fairly trivial.  Will that work for ya?

**Comment by gregvonkuster** [comment]
Ok, thanks.

**Comment by martenson** [comment]
eggs are smashed, this can probably continue

**Comment by bgruening** [comment]
We pushed the imaging datatypes and metadata a lot recently. I think we can close this one.
